### PR TITLE
[FEATURE] Implementation of ColumnPairValuesAGreaterThanB metric for PandasExecutionEngine

### PR DESCRIPTION
**Pull request by alexsherstinsky**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-56/GREAT-113, GREAT-56/GREAT-114, GREAT-56/GREAT-121, GREAT-56/GREAT-124
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [ ] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [ ] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/expectations/metrics/column_pair_map_metrics/column_pair_values_greater.py [hunk 1]
```diff
@@ -24,53 +17,26 @@ class ColumnPairValuesAGreaterThanB(MapMetricProvider):
...
+    def _pandas(cls, column_A, column_B, **kwargs):
+        allow_cross_type_comparisons = kwargs.get("allow_cross_type_comparisons")
         if allow_cross_type_comparisons:
             raise NotImplementedError
 
+        parse_strings_as_datetimes = kwargs.get("parse_strings_as_datetimes")
         if parse_strings_as_datetimes:
+            # noinspection PyPep8Naming
             temp_column_A = column_A.map(parse)
+            # noinspection PyPep8Naming
```
Comment (jcampbell) [inline great_expectations/expectations/metrics/column_pair_map_metrics/column_pair_values_greater.py:32]: @talagluck aren't we pulling that option out?

File: great_expectations/expectations/metrics/map_metric_provider.py [hunk 2]
```diff
@@ -730,6 +902,105 @@ def _pandas_column_map_condition_values(
...
+    domain_values = domain_values[boolean_mapped_unexpected_values == True]
+
+    result_format = metric_value_kwargs["result_format"]
+
+    if result_format["result_format"] == "COMPLETE":
+        return domain_values.to_dict("records")
+    else:
+        return domain_values[: result_format["partial_unexpected_count"]].to_dict(
+            "records"
+        )
```
Comment (jcampbell) [inline great_expectations/expectations/metrics/map_metric_provider.py]: Shouldn't this be something other than records based on the contents of the standard test?

**Review by jcampbell — CHANGES_REQUESTED** [review]
Shouldn't the cfe test now become enabled?

File: great_expectations/expectations/metrics/map_metric_provider.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/expectations/metrics/map_metric_provider.py]: @jcampbell You are correct.  This is fixed in a follow-on PR.  I will port the fix and re-commit.  This, in fact, resolves any issue with different output formats for the `unexpected_count` result.

**Review by NathanFarmer — APPROVED** [review]
Nothing for me to say except LGTM!

**Review by jcampbell — APPROVED** [review]
🎉

**Pull request merged**